src/main/java/com/igot/cb/masterdata/controller/MasterDataController.java

```diff
@@ -44,4 +48,37 @@ public ResponseEntity<ApiResponse> updateDegree(
         return new ResponseEntity<>(masterDataService.updateDegreesList(authToken, requestBody), HttpStatus.OK);
     }
 
+    @PostMapping(value = "/degree/search")
+    public ResponseEntity<ApiResponse> searchDegree(@RequestBody SearchCriteria searchCriteria) {
+        return new ResponseEntity<>(masterDataService.searchDegree(searchCriteria), HttpStatus.OK);
+    }
+
+    @PostMapping(value = "/institute/search")
+    public ResponseEntity<ApiResponse> searchInstitute(@RequestBody SearchCriteria searchCriteria) {
+        return new ResponseEntity<>(masterDataService.searchInstitute(searchCriteria), HttpStatus.OK);
+    }
+
+    @PostMapping(value = "/add/degree")
+    public ResponseEntity<ApiResponse> addDegree(@RequestBody Degree degree) {
+        return new ResponseEntity<>(masterDataService.addDegree(degree), HttpStatus.CREATED);
+    }
+
+    @PostMapping(value = "/add/institute")
+    public ResponseEntity<ApiResponse> addInstitute(@RequestBody Institute institute) {
+        return new ResponseEntity<>(masterDataService.addInstitute(institute), HttpStatus.CREATED);
+    }
+
+    @PutMapping("/degree/update/status")
+    public ResponseEntity<ApiResponse> updateDegreeStatus(@RequestBody StatusUpdateRequest request) {
+        ApiResponse response = masterDataService.toggleDegreeStatusByName(request.getName(), request.getStatus());
+        return ResponseEntity.status(response.getParams().getStatus().equalsIgnoreCase(Constants.SUCCESS) ? HttpStatus.OK : HttpStatus.BAD_REQUEST).body(response);
+    }
+
+    @PutMapping("/institute/update/status")
+    public ResponseEntity<ApiResponse> instituteUpdateStatus(@RequestBody StatusUpdateRequest request) {
+        ApiResponse response = masterDataService.toggleInstituteStatusByName(request.getName(), request.getStatus());
+        return ResponseEntity.status(response.getParams().getStatus().equalsIgnoreCase(Constants.SUCCESS) ? HttpStatus.OK : HttpStatus.BAD_REQUEST).body(response);
+    }
```

Comment:
the HttpStatus value must be set from the apiResponse object which is returned from the service calls